### DISCUSSION TITLE
Docker dependencies on Quay breaks after a while

### DISCUSSION THREAD

**Discussion by ducretje-evooq** [renovatebot/renovate#21599] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

--

### If you're self-hosting Renovate, select which platform you are using.

None

### If you're self-hosting Renovate, tell us what version of the platform you run.

--

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

We want to have `renovate` able to get the new tags from Quay.io private registries.

### What you tried so far.

When adding a hostRules block to handle private repositories on Quay.io:

    {
      "hostType": "docker",
      "matchHost": "https://quay.io",
      "username": "<REDACTED>",
      "encrypted": {
        "token": "<REDACTED>"
    }

We create an OAuth token on quay.
The token is encrypted using [renovate encryption tool](https://app.renovatebot.com/encrypt).

renovate works for a while (~1 day), no error / warnings in the logs.
After ~1 day we get an error in the dashboard:

<img width="449" alt="image" src="https://user-images.githubusercontent.com/70567885/233022960-d3c12122-6238-4f1c-80c1-55c5a7b7ea94.png">

And the logs show a 401 error on quay

The token is valid, if we try to curl the quay's API it works

```sh
curl -s -X GET -H "Authorization: Bearer <REDACTED>" "https://quay.io/api/v1/repository/<REDACTED>/tag/" 
```

But we can see in the logs that renovate actually uses `api/v2/`

### Re-encrypt the same token

If we re-encrypt the same token, we get a different hash which is expected, and it fixes the issue for one additional day, then it fails again.

### Use user:token

We tried to use the CLI credentials

```json
    {
      "hostType": "docker",
      "matchHost": "https://quay.io",
      "username": "<REDACTED>",
      "encrypted": {
        "password": "<REDACTED>"
      }
    },
```

> 🛑 Got a 401

We tried to use the `token` parameter instead of `password`

```json
    {
      "hostType": "docker",
      "matchHost": "https://quay.io",
      "username": "<REDACTED>",
      "encrypted": {
        "token": "<REDACTED>"
      }
    },
```

> 🛑 Got a 401

We also notice that this time, renovate calls `api/v1/`, which is weird since we have the same syntax than when using the OAuth token.

We can't create a reproduction repository since I won't provide access to private repositories.

### Relevant debug logs

<details><summary>Logs</summary>

Using an OAuth token after a while

```
DEBUG: GET https://quay.io/v2/auth?service=quay.io&scope=repository:<REDACTED>:pull = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=401 retryCount=0, duration=264)
WARN: No docker auth found - returning
```

Using CLI credentials

```
DEBUG: GET https://quay.io/api/v1/repository/<REDACTED>/tag/?limit=100&page=1&onlyActiveTags=true = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=401 retryCount=0, duration=972)
DEBUG: Datasource unauthorized
```

</details>

**Comment by rarkins** [comment]
Renovate emits a list of http URLs it tries at the end of each run. Can you find and compare the list of quay.io URLs during a successful run to the list during an unsuccessful?
